src/init.js

```diff
@@ -10,21 +10,29 @@ const isMobileDevice = typeof window !== 'undefined' && userAgentParser.getDevic
 const isTabletDevice = typeof window !== 'undefined' && userAgentParser.getDevice().type === 'tablet';
 
 // initialise flags with default values in ld redux store
-const initFlags = (flags, dispatch) => {
+const initFlags = (flags, dispatch, camelCased = true) => {
   const flagValues = { isLDReady: false };
   for (const flag in flags) {
-    const camelCasedKey = camelCase(flag);
-    flagValues[camelCasedKey] = flags[flag];
+    if (camelCased) {
+      const camelCasedKey = camelCase(flag);
+      flagValues[camelCasedKey] = flags[flag];
+    } else {
+      flagValues[flag] = flags[flag];
+    }
   }
   dispatch(setFlagsAction(flagValues));
 };
 
 // set flags with real values from ld server
-const setFlags = (flags, dispatch) => {
+const setFlags = (flags, dispatch, camelCased = true) => {
   const flagValues = { isLDReady: true };
   for (const flag in flags) {
-    const camelCasedKey = camelCase(flag);
-    flagValues[camelCasedKey] = ldClient.variation(flag, flags[flag]);
+    if (camelCased) {
+      const camelCasedKey = camelCase(flag);
+      flagValues[camelCasedKey] = ldClient.variation(flag, flags[flag]);
+    } else {
+      flagValues[flag] = flags[flag];
```

Comment:
The flag value here should be resolved using the `ldClient.variation` method because it performs some analytics and possibly other logic that are useful. I have also suggested some refactoring below for dry purposes:

```js
const setFlags = (flags, dispatch, useCamelCaseFlagKeys) => {
  const flagValues = { isLDReady: true };
  for (const flag in flags) {
    const flagKey = useCamelCaseFlagKeys ? camelCase(flag) : flag;
    flagValues[flagKey] = ldClient.variation(flag, flags[flag]);
  }
  dispatch(setFlagsAction(flagValues));
};
```